intermediary_representation.py

```diff
@@ -216,13 +216,33 @@ def __init__(self, scene: bpy.types.Scene, root: Node = Node()):
         self.root = root
         self.scene = scene
 
-    def add_node(self, o: bpy.types.Object, parent: Node = None) -> Node:
+    def add_node(self, o: bpy.types.Object = None, parent: Node = None) -> Node:
         log.debug(f"Adding Blender Object: {o} as a node for root {self.root} and {parent} as parent")
 
-        node = None
+        node = self._translate(o) if o else Node('Placeholder node')
+
+        if self.root is None:
+            self.root = node
+        else:
+            node_parent = parent if parent else self._resolve_parenting(o)
```

Comment:
Can the call to _resolve_parenting fail?

Reply:
hi there Gunnar!

Actually, now that you have said so, it can create a loop if mishandled.

For flexibility reasons, not every node has to be mapped 1:1 to a Blender object and these nodes without a counterpart in Blender were called 'placeholder' nodes. That is the point of having this intermediary representation in the first place : it exists so we can freely adapt Blender's internal scene representation in ways that make more sense to the exporter.

Now if we consider the case in which somebody creates a placeholder node and add it as root - which we actually do, for instance a placeholder node is created to serve as a handle for the IR scene graph - then this will likely work out this way:

1. find_from_blender_object will hopefully return None because it cannot find a node if we don't provide a Blender object.
2. thus self.root will be returned, in which case a self reference will be created.

And i am not 100% sure about (1) to be honest. 

I should an assert for (1). What do you think about (2)?